cassandra-bridge/src/main/java/org/apache/cassandra/spark/utils/TimeProvider.java

```diff
@@ -19,16 +19,37 @@
 
 package org.apache.cassandra.spark.utils;
 
+import java.util.concurrent.TimeUnit;
+
 /**
  * Provides current time
   */
-@FunctionalInterface
 public interface TimeProvider
 {
-    TimeProvider INSTANCE = () -> (int) Math.floorDiv(System.currentTimeMillis(), 1000L);
+    TimeProvider DEFAULT = new TimeProvider()
+    {
+        private final int referenceEpoch = nowInSeconds();
```

Comment:
I worry this will be a gotcha for those unfamiliar with `referenceEpoch` and why it is required. Many users will not realize they need to override TimeProvider.

If they have long running Spark clusters, this `referenceEpoch` will be defined once and never updated, potentially resulting in values appearing that should be been TTL'ed

Reply:
My thinking is that SBR reads a snapshot. Ideally, whatever is unexpired at the moment of taking snapshot, should be read back, making it "snapshot-isolated-ish". Using a fixed during the job can archive it, especially the job creates a snapshot. 
What is the "long running spark cluster"? Is it long running because the _one_ snapshot has a large amount of data?

Reply:
maybe we can preserve existing behavior by returning the current seconds every time.

```
        @Override
         public int referenceEpochInSeconds()
         {
             return nowInSeconds();
         }
```

One clarification, SBR when running with the CassandraDataLayer, reads a snapshot. The same is not true for when you are using the `LocalDataLayer` for example.

Reply:
The reason of using a single time point is applicable to SBR in general.
The expectation of bulk read is no different than a single point read. It is surprising if data visibility changes (caused by the advancing time) during the read. 

Bulk read is essentially a transformation from a collection of sstable to a collection of mutations. The time reference has to be the same in order to get the same collection of mutations deterministically. The existing behavior is a bug, especially when TTL is considered. It has little to do with test setup or prod use case.